mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergSerDe.java

```diff
@@ -56,10 +62,22 @@ public void initialize(@Nullable Configuration configuration, Properties serDePr
     } else if (serDeProperties.get(InputFormatConfig.TABLE_SCHEMA) != null) {
       tableSchema = SchemaParser.fromJson((String) serDeProperties.get(InputFormatConfig.TABLE_SCHEMA));
     } else {
-      try {
-        tableSchema = Catalogs.loadTable(configuration, serDeProperties).schema();
-      } catch (NoSuchTableException nte) {
-        throw new SerDeException("Please provide an existing table or a valid schema", nte);
+      // Read the configuration parameters
+      String columnNames = serDeProperties.getProperty(serdeConstants.LIST_COLUMNS);
+      String columnTypes = serDeProperties.getProperty(serdeConstants.LIST_COLUMN_TYPES);
+      String columnNameDelimiter = serDeProperties.containsKey(serdeConstants.COLUMN_NAME_DELIMITER) ?
+          serDeProperties.getProperty(serdeConstants.COLUMN_NAME_DELIMITER) : String.valueOf(SerDeUtils.COMMA);
+      if (columnNames != null && columnTypes != null && columnNameDelimiter != null &&
+          !columnNames.isEmpty() && !columnTypes.isEmpty() && !columnNameDelimiter.isEmpty()) {
+        tableSchema = HiveSchemaUtil.schema(columnNames, columnTypes, columnNameDelimiter);
```

Comment:
Does this preserve field name casing or will the return schema have all lowercase field names?

Reply:
Hive always converts the column/table names to lower case.
Updated the test case to contain some capital letters to test/confirm this.

Good to have a test case for this as well.
Thanks!